config/prod/minidream-R-env-pub-prod-instance-ec2.yaml

```diff
@@ -0,0 +1,46 @@
+# Provision an EC2 instance
+template_path: remote/managed-ec2.yaml
+stack_name: minidream-R-env-pub-prod-instance-ec2 
+parameters:
+  # The Sage deparment (Platform, CompOnc, SysBio, Governance, etc..)
+  Department: "Computational Oncology"
+  # The Sage project this resource will be used for
+  Project: "CSBC"
+  OwnerEmail: "milen.nikolov@sagebase.org"
+
+  # Default account settings, leave alone unless you plan to override
+  VpcName: "{{stack_group_config.default_vpc}}"
+  KeyName: "{{stack_group_config.default_key_pair}}"
+  AMIId: "{{stack_group_config.default_ami}}"
+
+  # Settings have default values but can be overriden. Set the below
+  # parameters *only* if you want to override the defaults.
+
+  # (Optional) EC2 instance type, default is "t2.nano" (other available types https://aws.amazon.com/ec2/instance-types/)
+  InstanceType: "r4.16xlarge"
+  # (Optional) EC2 boot volume size in GB, default is 8GB, Max is 1000GB
+  VolumeSize: "500"
+  # (Optional) Base instance on AMI (default: ami-082278746f893d99c) (supported distros: AWS linux)
+  # AMIId: "ami-0de53d8956e8dcf80"
+  # (Optional) Run EC2 in a subnet (default: PrivateSubnet) (valid values: PrivateSubnet or PublicSubnet)
+  VpcSubnet: "PublicSubnet"
```

Comment:
Due to new security regulation we don't want to allow any public instances on scicomp.  Please comment this out so that it's private.  We will add a load balancer for your app so that users can reach it.  We will also setup a SSL certificate so your app can be reached with `https` instead of `http`

Reply:
Hmm, I thought we discussed that with @ahayden and @xschildw yesterday. Initially (yesterday) that was the reason to go with sandbox, if I understand correctly. Today @ahayden said that scicomp is segmented enough, and we can proceed with having public ip and a web server running there (or maybe I'm misunderstanding)? The public ip is crucial for this project even before the class start, so that external people can access the environment. If scicomp doesn't allow public ip then it's a deal breaker, and I'd have to provision an amazon instance differently. If that's the case, do we have a list of the most relevant/significant issues of running the instance on sandbox (@zaro0508 mentioned accidental deletion, which we definitely do not want; what are other issues?). If the issues are too prohibitive we would probably have to start provision ad hoc outside scicomp and sandbox....

Reply:
Don't worry, your users will be able to access your application, instead of serving your app from the machine directly we will serve your app up from a load balancer which adds an additional level of security.   Making the instance private is to prevent an attacker from being able to directly access the machine itself.